MelonLoader/MelonEvents.cs

```diff
@@ -0,0 +1,91 @@
+namespace MelonLoader
+{
+    public static class MelonEvents
+    {
+        /// <summary>
+        /// Occurs after all MelonPlugins are initialized.
+        /// </summary>
+        public readonly static MelonEvent OnPreInitialization = new MelonEvent(true);
+
+        /// <summary>
+        /// Occurs before the Start Screen is loaded.
+        /// </summary>
+        public readonly static MelonEvent OnApplicationEarlyStart = new MelonEvent(true);
+
+        /// <summary>
+        /// Occurs after all MelonMods are initialized and before the right Support Module is loaded.
+        /// </summary>
+        public readonly static MelonEvent OnPreSupportModule = new MelonEvent(true);
+
+        /// <summary>
+        /// Occurs after all MelonLoader components are fully initialized (including all MelonMods).
+        /// <para>Don't use this event to initialize your mods anymore! Instead, override <see cref="MelonBase.OnInitializeMelon"/>.</para>
+        /// </summary>
+        public readonly static MelonEvent OnApplicationStart = new MelonEvent(true);
+
+        /// <summary>
+        /// Occurs when the Engine is fully initialized.
+        /// </summary>
+        public readonly static MelonEvent OnApplicationLateStart = new MelonEvent(true);
+
+        /// <summary>
+        /// Occurs when... well, the name pretty much says it.
```

Comment:
A little more professional here, please. Also, please change all "Occurs when" to "Called when"

Reply:
Damn it lmao. Fine

Reply:
Also, they all start with "Occurs" cuz they're all events, but if you really mind it that much then fine

Reply:
https://github.com/SlidyDev/MelonLoader/commit/e25c708b3ec3d15da00b057285b4b15dae17f004